src/kernels/MIOpenTensorKernelsHip.cpp

```diff
@@ -91,4 +91,87 @@ extern "C" __global__ void Op1dTensorGeneric(const MIOPEN_TYPE* a,
     }
 }
 
+#endif
+
+#ifdef USE_2D_TENSOR_GENERIC
+// NC
+extern "C" __global__ void Op2dTensorGeneric(MIOPEN_TYPE* a,
+                                             const int a_nstride,
+                                             MIOPEN_TYPE* b,
+                                             const int b_c,
+                                             const int b_nstride,
+                                             MIOPEN_TYPE* c,
+                                             const int c_c,
+                                             const int c_nstride,
+                                             const MIOPEN_TYPE alpha0,
+                                             const MIOPEN_TYPE alpha1,
+                                             const MIOPEN_TYPE beta,
+                                             const unsigned int bitmap,
+                                             const int work_per_wg,
+                                             const long Aoffset,
+                                             const long Boffset,
+                                             const long Coffset,
+                                             const int num_wg)
+{
+    int gid = blockIdx.x;
+
+    MIOPEN_TYPE* a_off = a + Aoffset;
+    MIOPEN_TYPE* b_off = b + Boffset;
+    MIOPEN_TYPE* c_off = c + Coffset;
+
+    int o_n_div = bitmap & (1 << 0) ? 1 : c_c;
+
+    // num_wg: the number of workgroups should be launched
+    // MAX_NUM_WG: the maximum number of workgroups actually launched
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wfloat-equal"
+    if(beta == (MIOPEN_TYPE)0)
```

Comment:
Please avoid C-style casts, use proper `static_cast` instead.
It won't pass CI in any case.